data/transactions/verify/streamverifier.go

```diff
@@ -173,66 +96,57 @@ func (sv *StreamVerifier) WaitForStop() {
 	sv.activeLoopWg.Wait()
 }
 
-func (sv *StreamVerifier) cleanup(pending []*UnverifiedElement) {
-	// report an error for the unchecked txns
-	// drop the messages without reporting if the receiver does not consume
-	for _, uel := range pending {
-		sv.sendResult(uel.TxnGroup, uel.BacklogMessage, errShuttingDownError)
-	}
-}
-
 func (sv *StreamVerifier) batchingLoop() {
 	defer sv.activeLoopWg.Done()
-	timer := time.NewTicker(waitForNextTxnDuration)
+	timer := time.NewTicker(waitForNextElmtDuration)
 	defer timer.Stop()
 	var added bool
 	var numberOfSigsInCurrent uint64
 	var numberOfBatchAttempts uint64
-	ue := make([]*UnverifiedElement, 0, 8)
-	defer func() { sv.cleanup(ue) }()
+	uElmts := make([]UnverifiedElement, 0, 8)
+	defer func() { sv.ep.Cleanup(uElmts, ErrShuttingDownError) }()
 	for {
 		select {
-		case stx := <-sv.stxnChan:
-			numberOfBatchableSigsInGroup, err := getNumberOfBatchableSigsInGroup(stx.TxnGroup)
+		case elem := <-sv.inputChan:
+			numberOfBatchableSigsInGroup, err := elem.GetNumberOfBatchableSigsInGroup()
 			if err != nil {
-				// wrong number of signatures
-				sv.sendResult(stx.TxnGroup, stx.BacklogMessage, err)
+				sv.ep.GetErredUnverified(elem, err)
 				continue
 			}
 
 			// if no batchable signatures here, send this as a task of its own
 			if numberOfBatchableSigsInGroup == 0 {
-				err := sv.addVerificationTaskToThePoolNow([]*UnverifiedElement{stx})
+				err := sv.addVerificationTaskToThePoolNow([]UnverifiedElement{elem})
 				if err != nil {
 					return
 				}
-				continue // stx is handled, continue
+				continue // elem is handled, continue
 			}
 
-			// add this txngrp to the list of batchable txn groups
+			// add this element to the list of batchable elements
 			numberOfSigsInCurrent = numberOfSigsInCurrent + numberOfBatchableSigsInGroup
-			ue = append(ue, stx)
+			uElmts = append(uElmts, elem)
 			if numberOfSigsInCurrent > txnPerWorksetThreshold {
```

Comment:
Why `txnPerWorksetThreshold` is in txn.go and not in streamverifeir.go:
`txnPerWorksetThreshold` existed before this project, and is used for non-streamed batching (block verification) outside the streamverifier.go

Why using the same parameter here: 
Since I don't have a justification to create another parameter or modify its value, I am using it in streamverifier.go